nomad-base/src/settings/secrets.rs

```diff
@@ -0,0 +1,67 @@
+//! Secrets configuration for agents.
+//!
+//! This struct is serialized from a JSON file or built drawing from a hosted
+//! secrets manager backend. This struct is then used to finish building an
+//! agents `Settings` block (see settings/mod.rs) along with a `NomadConfig`.
+//!
+//! Example JSON File Format
+//! {
+//!     "rpcs": {
+//!         "ethereum": {
+//!             "rpcStyle": "ethereum",
+//!             "connection": {
+//!                 "type": "http",
+//!                 "url": ""
+//!             }
+//!         },
+//!         "moonbeam": {
+//!             "rpcStyle": "ethereum",
+//!             "connection": {
+//!                 "type": "http",
+//!                 "url": ""
+//!             }
+//!         },
+//!     },
+//!     "transactionSigners": {
+//!         "ethereum": {
+//!             "type": "hexKey"
+//!             "key": "",
+//!         },
+//!         "moonbeam": {
+//!             "type": "hexKey"
+//!             "key": "",
+//!         },
+//!     },
+//!     "attestationSigner": {
+//!         "key": "",
+//!         "type": "hexKey"
+//!     }
+//! }
+
+use color_eyre::Report;
+use nomad_xyz_configuration::{agent::SignerConf, ChainConf};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::{fs::File, io::BufReader, path::Path};
+
+/// Agent secrets block
+#[derive(Debug, Deserialize, Default, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AgentSecrets {
+    /// RPC endpoints
+    pub rpcs: HashMap<String, ChainConf>,
+    /// Transaction signers
+    pub transaction_signers: HashMap<String, SignerConf>,
+    /// Attestation signers
+    pub attestation_signer: Option<SignerConf>,
+}
+
+impl AgentSecrets {
```

Comment:
1. Will add a validate method to ensure every key has a non-empty value for rpcs and tx signers
2. Watcher and updater will also check attestation signer is some in validation method for base settings